galley/pkg/runtime/states/defaultstate/state.go

```diff
@@ -73,21 +72,22 @@ type resourceTypeState struct {
 	versions map[resource.FullName]resource.Version
 }
 
-func newState(name string, schema *resource.Schema, cfg *Config, strategy *publishingStrategy,
-	distributor Distributor) *State {
+// New creates a new State
+func New(name string, domainSuffix string, schema *resource.Schema, strategy *publish.Strategy,
+	distributor publish.Distributor) *State {
 
 	now := time.Now()
 	s := &State{
 		name:             name,
+		domainSuffix:     domainSuffix,
```

Comment:
Are you replacing config with domainSuffix due to cyclic dependency problems?

Reply:
no, just noticed only a part was needed ... reverted.